src/rgw/rgw_op.cc

```diff
@@ -2716,22 +2715,9 @@ void RGWStatBucket::execute()
     return;
   }
 
-  RGWUserBuckets buckets;
-  bucket.bucket = s->bucket;
-  buckets.add(bucket);
-  map<string, RGWBucketEnt>& m = buckets.get_buckets();
-  op_ret = store->getRados()->update_containers_stats(m);
-  if (! op_ret)
-    op_ret = -EEXIST;
-  if (op_ret > 0) {
-    op_ret = 0;
-    map<string, RGWBucketEnt>::iterator iter = m.find(bucket.bucket.name);
-    if (iter != m.end()) {
-      bucket = iter->second;
-    } else {
-      op_ret = -EINVAL;
-    }
-  }
+  rgw::sal::RGWRadosUser user(store, s->user->user_id);
+  bucket = new rgw::sal::RGWRadosBucket(store, user, s->bucket);
```

Comment:
what's happening here--operating directly on RGWRadosBucket?

Reply:
Bucket's not done yet.  You can't instantiate RGWSalBucket (pure virtual), so we have to create a RGWRadosBucket at this point.